__tests__/dotnet.test.ts

```diff
@@ -21,9 +23,11 @@ describe('When disabling the .NET Core telemetry', () => {
 
 describe('When successfully installing the Cake Tool locally', () => {
   const fakeExec = exec as jest.MockedFunction<typeof exec>;
+  const fakeFsExistsSync = fs.existsSync as jest.MockedFunction<typeof fs.existsSync>;
```

Comment:
Here, you're faking `fs` which is a [transitive dependency](https://en.wikipedia.org/wiki/Transitive_dependency) of the _System Under Test_ (`SUT` → `ToolsDirectory` → `fs`). If the implementation of `ToolsDirectory` would ever to change to use a different API (e.g. [`fs.accessSync`](https://nodejs.org/api/fs.html#fs_fs_accesssync_path_mode)), the tests would break. I think it would be better to follow the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) here and mock the `ToolsDirectory.containsFile` function, instead.

This will make the tests pass even if there's a file named `tools/dotnet-cake` on the file system:

```ts
beforeAll(() => {
  ToolsDirectory.prototype.containsFile = jest.fn().mockImplementation(() => {
    return false;
  });
});
```

Reply:
Agreed, updated PR.

Reply:
Awesome! Could you please also update the commit message in c017e1d?

Reply:
done.